### PR TITLE
Fix volume mode lag by yielding during base64 and gzip encoding

### LINKED ISSUE #2136: Volume Trace mode lags sometimes
See https://discuss.webknossos.org/t/volume-mode-bug/585

<a href="https://timer.scm.io/repos/537237ce1a00001a003c4a14/issues/2136/create?referer=github" target="_blank">Log Time</a>

### PR DESCRIPTION
**Pull request by daniel-wer**
This PR solves the volume mode save lag, by yielding during the base64 encoding of buckets and the gzip encoding of the update actions, so the main thread is not blocked by these performance heavy computations.

For some reason in the `sendToStoreImpl` I couldn't use `await Promise.all(batch.map(async ...))` because Flow complained that the result was not the array it expected :confused: 

### Mailable description of changes:
 - Fixed lag that periodically occurred in volume mode.

### Steps to test:
- Use the `Trace` mode in a volume tracing. There should be no lag when tracing immediately after the last slice was traced and there should also be no lag during the saving of the save queue (trigger using `Ctrl/Cmd + Save`).
- Saving should not take overly long as a result of this PR

### Issues:
- fixes #2136

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/model/binary/layers/wk_layer.js [hunk 1]
```diff
@@ -83,11 +84,17 @@ class WkLayer extends Layer {
...
 
   async sendToStoreImpl(batch: Array<DataBucket>): Promise<void> {
-    const items = batch.map(bucket => {
+    let counter = 0;
+    const items = [];
+    for (const bucket of batch) {
+      counter++;
+      // Do not block the main thread for too long as Base64.fromByteArray is performance heavy
+      // eslint-disable-next-line no-await-in-loop
+      if (counter % 3 === 0) await Utils.sleep(1);
```
Comment (normanrz) [inline app/assets/javascripts/oxalis/model/binary/layers/wk_layer.js]: Can you make `3` a named constant? 
How did you come up with that? Did you measure how long it takes to encode one bucket?

**Comment by normanrz** [comment]
I am having the issue that some saves don't go through and the the save icon never turns to checkmark (even when clicked).
bildschirmfoto 2017-11-16 um 13 21 37

How to reproduce:
* Open a tracing
* Trace a large section
* Move to next slide
* Repeat 10 times
* Click on Save
* Wait for godot

File: app/assets/javascripts/oxalis/model/binary/layers/wk_layer.js (hunk 1 above)
Comment (daniel-wer) [reply app/assets/javascripts/oxalis/model/binary/layers/wk_layer.js]: Yes. Encoding three buckets takes ~15ms on average and no lag is noticeable.

**Comment by daniel-wer** [comment]
Nice catch! I found the race condition during saving and fixed it, see my commit and comment in there.

**Pull request merged**